sstables/sstables.hh

```diff
@@ -679,6 +679,14 @@ private:
 
     future<> create_data() noexcept;
 
+    // Methods to calculate the total memory used by the components and to
+    // free them, if required, to reclaim memory back to the system.
+    // Note that both these methods right now only consider bloom filters
+    // for calculating the total memory used and for reclamation.
+    // TODO: Support discarding other in-memory components.
+    uint64_t total_components_memory_size() const;
+    uint64_t reclaim_memory_from_components();
```

Comment:
As soon as we add support for discarding other components which can participate in reads (summary, compression-info), we will have to futurize this. But we can  wait with that until we actually get there.

Reply:
Ack.